lib/lints/avoid_late_keyword/avoid_late_keyword_rule.dart

```diff
@@ -31,8 +33,14 @@ class AvoidLateKeywordRule extends SolidLintRule {
     CustomLintContext context,
   ) {
     context.registry.addVariableDeclaration((node) {
-      if (node.declaredElement?.isLate ?? false) {
+      final initializedDynamically = config.parameters.initializedDynamically;
+      final isLateDeclaration = node.declaredElement?.isLate ?? false;
+      final hasInitializer = node.initializer != null;
+
+      if ((!initializedDynamically || !hasInitializer) && isLateDeclaration) {
```

Comment:
This whole bit can be simplified.

Let's try to write it out as boring as we can.

We're gonna extract a private function and decompose the statement to reflect how we actually think about the problem:

```dart
bool _shouldLint(dynamic node) {
  final isLateDeclaration = node.declaredElement?.isLate ?? false;
  if (!isLateDeclaration) return false;
  
  final allowInitialized = config.parameters.allowInitialized;
  if (!allowInitialized) return true; // all late`s are linted
  
  return !hasInitializer;
}
```

This will make the code more easy to extend, and show clear precedence of rules. No mental math needed to understand what's happening.